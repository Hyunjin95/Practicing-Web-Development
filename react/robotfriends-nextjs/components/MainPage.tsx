import React from 'react';
import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundary from './ErrorBoundary';
import Header from './Header';

import { IRobot, IAppProps } from '../pages/index';

interface IStates {
    robots: Array<IRobot>,
    searchField: string,
}

class MainPage extends React.Component<IAppProps, IStates> {
    constructor(props: IAppProps) {
        super(props);
        this.state = {
            robots: props.robots,
            searchField: ''
        };
    }

    filteredRobots = (): Array<IRobot> => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.state.searchField.toLowerCase());
        });
    };

    onSearchChange = (event: React.SyntheticEvent<HTMLInputElement>): void => {
        this.setState({ searchField: event.currentTarget.value })
    }

    render() {
        const { isError } = this.props;
        
        return!this.state.robots.length ?
            <h1>Loading...</h1>
            :
            (
            <div className="tc">
                <Header />
                <SearchBox searchChange={this.onSearchChange} />
                <Scroll>
                    <ErrorBoundary isError={isError}>
                        <CardList robots={this.filteredRobots()} />
                    </ErrorBoundary>
                </Scroll>    
            </div>
            );
    }
}

export default MainPage;