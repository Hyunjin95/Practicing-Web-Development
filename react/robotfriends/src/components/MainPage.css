@import url('https://fonts.googleapis.com/css?family=Luckiest+Guy');

h1 {
    font-display: swap;
    font-family: 'Luckiest Guy', cursive;
    font-size: 5em;
    color: #0ccac4;
    font-weight: 200;
}

input {
    width: 100%;
    max-width: 400px;
    box-sizing: border-box;
}