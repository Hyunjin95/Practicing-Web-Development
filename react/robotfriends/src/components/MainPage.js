import React from 'react';
import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundary from './ErrorBoundary';
import Header from './Header';
import './MainPage.css';

class MainPage extends React.Component {
    componentDidMount() {
        console.log(process.env.REACT_APP_SAY_HI);
        this.props.onRequestRobots();
    }

    filteredRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        });
    };

    render() {
        const { onSearchChange, isPending, isError } = this.props;
        
        if (isPending) {
            return <h1 className='tc f1'>Loading</h1>;
        }
        else {
            return (
                <div className="tc">
                    <Header />
                    <SearchBox searchChange={onSearchChange} />
                    <Scroll>
                        <ErrorBoundary isError={isError}>
                            <CardList robots={this.filteredRobots()} />
                        </ErrorBoundary>
                    </Scroll>    
                </div>
            );
        }
    }
}

export default MainPage;