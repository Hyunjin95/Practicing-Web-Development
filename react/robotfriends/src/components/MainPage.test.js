import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

it('expect to render App container', () => {
    const mockStore = {
        robots: [],
        searchField: ''
    };
    expect(shallow(<MainPage store={mockStore} />).debug()).toMatchSnapshot();
});