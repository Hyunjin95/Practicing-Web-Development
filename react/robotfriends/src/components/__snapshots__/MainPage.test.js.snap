// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
"<div className=\\"tc\\">
  <Header />
  <SearchBox searchChange={[undefined]} />
  <Scroll>
    <ErrorBoundary isError={false}>
      <CardList robots={{...}} />
    </ErrorBoundary>
  </Scroll>
</div>"
`;

exports[`shows pending messages when pennding 1`] = `
"<h1 className=\\"tc f1\\">
  Loading
</h1>"
`;
